File: ui_v2/frontend/src/types/item.ts
/** OSRS item as returned by the API. */
export interface Item {
  id: number;
  name: string;
  tradeable_on_ge: boolean;
  members: boolean;
  noted: boolean;
  noteable: boolean;
  placeholder?: boolean;
  stackable: boolean;
  equipable: boolean;
  cost: number;
  lowalch: number;
  highalch: number;
  icon_b64?: string | null;
}

export interface ItemSearchResponse {
  success: boolean;
  results: Item[];
  count: number;
}

export interface ItemDetailResponse {
  success: boolean;
  item?: Item;
  error?: string;
}

File: ui_v2/backend/models/item.py
"""Pydantic models for item search and detail responses."""

from typing import Optional, List
from pydantic import BaseModel


class ItemModel(BaseModel):
    id: int
    name: str
    tradeable_on_ge: bool
    members: bool
    noted: bool
    noteable: bool
    placeholder: Optional[bool] = None
    stackable: bool
    equipable: bool
    cost: int
    lowalch: int
    highalch: int
    icon_b64: Optional[str] = None


class ItemSearchResponse(BaseModel):
    success: bool
    results: List[ItemModel] = []
    count: int = 0


class ItemDetailResponse(BaseModel):
    success: bool
    item: Optional[ItemModel] = None
    error: Optional[str] = None
